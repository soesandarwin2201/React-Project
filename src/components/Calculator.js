/*eslint-disable*/
import React from 'react';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
class Claculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.eventHandler = this.eventHandler.bind(this);
  }

  eventHandler(e) {
    this.setState(calculate(this.state, e.target.textContent));
  }

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className='cal-grid'>
        <div className='cal-result'>
          <p>{total}</p>
          <p>{operation}</p>
          <p>{next}</p>
        </div>
        <div className='cal-btn-body'>
          <div className='row'>
            <div onClick={this.eventHandler} className='col-1 col'>
              AC
            </div>
            <div onClick={this.eventHandler} className='col-2 col'>
              +/-
            </div>
            <div onClick={this.eventHandler} className='col-3 col'>
              %
            </div>
            <div onClick={this.eventHandler} className='col-4 col'>
              รท
            </div>
          </div>
          <div className='row'>
            <div onClick={this.eventHandler} className='col-1 col'>
              7
            </div>
            <div onClick={this.eventHandler} className='col-2 col'>
              8
            </div>
            <div onClick={this.eventHandler} className='col-3 col'>
              9
            </div>
            <div onClick={this.eventHandler} className='col-4 col'>
              x
            </div>
          </div>
          <div className='row'>
            <div onClick={this.eventHandler} className='col-1 col'>
              4
            </div>
            <div onClick={this.eventHandler} className='col-2 col'>
              5
            </div>
            <div onClick={this.eventHandler} className='col-3 col'>
              6
            </div>
            <div onClick={this.eventHandler} className='col-4 col'>
              -
            </div>
          </div>
          <div className='row'>
            <div onClick={this.eventHandler} className='col-1 col'>
              1
            </div>
            <div onClick={this.eventHandler} className='col-2 col'>
              2
            </div>
            <div onClick={this.eventHandler} className='col-3 col'>
              3
            </div>
            <div onClick={this.eventHandler} className='col-4 col'>
              +
            </div>
          </div>
          <div className='row'>
            <div onClick={this.eventHandler} className='col-1 span col'>
              0
            </div>
            <div onClick={this.eventHandler} className='col-3 col'>
              .
            </div>
            <div onClick={this.eventHandler} className='col-4 col'>
              =
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Claculator;
