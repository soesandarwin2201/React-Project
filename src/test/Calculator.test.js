import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

it('render correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
