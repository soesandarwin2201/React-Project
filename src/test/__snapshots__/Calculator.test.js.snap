// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<div
  className="app"
>
  <div
    className="calculator-title"
  >
    <h1>
      Let's do some math!
    </h1>
  </div>
  <div
    className="cal-grid"
  >
    <div
      className="cal-result"
    >
      <p>
        0
      </p>
      <p />
      <p />
    </div>
    <div
      className="cal-btn-body"
    >
      <div
        className="row"
      >
        <div
          className="col-1 col"
          onClick={[Function]}
        >
          AC
        </div>
        <div
          className="col-2 col"
          onClick={[Function]}
        >
          +/-
        </div>
        <div
          className="col-3 col"
          onClick={[Function]}
        >
          %
        </div>
        <div
          className="col-4 col"
          onClick={[Function]}
        >
          รท
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1 col"
          onClick={[Function]}
        >
          7
        </div>
        <div
          className="col-2 col"
          onClick={[Function]}
        >
          8
        </div>
        <div
          className="col-3 col"
          onClick={[Function]}
        >
          9
        </div>
        <div
          className="col-4 col"
          onClick={[Function]}
        >
          x
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1 col"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="col-2 col"
          onClick={[Function]}
        >
          5
        </div>
        <div
          className="col-3 col"
          onClick={[Function]}
        >
          6
        </div>
        <div
          className="col-4 col"
          onClick={[Function]}
        >
          -
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1 col"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="col-2 col"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="col-3 col"
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="col-4 col"
          onClick={[Function]}
        >
          +
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-1 span col"
          onClick={[Function]}
        >
          0
        </div>
        <div
          className="col-3 col"
          onClick={[Function]}
        >
          .
        </div>
        <div
          className="col-4 col"
          onClick={[Function]}
        >
          =
        </div>
      </div>
    </div>
  </div>
</div>
`;
